<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>460</width>
    <height>780</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Packet Storm</string>
  </property>
  <property name="styleSheet">
   <string notr="true">
   /* =======================================================
      GLOBAL SETTINGS (全局设置)
      强制所有控件使用 JetBrains Mono，营造终端感
      ======================================================= */
   * {
       font-family: "JetBrains Mono", "Consolas", monospace;
       font-size: 13px;
       color: #a0a8b7; /* 默认文字颜色：灰白，避免刺眼 */
   }

   QMainWindow, QWidget {
       background-color: #0b0f19; /* 极深蓝黑色背景 */
   }

   /* =======================================================
      GROUP BOX (分组框 - 模块化设计)
      科技感的细边框，青色标题
      ======================================================= */
   QGroupBox {
       border: 1px solid #1c2333; /* 暗淡的边框 */
       border-radius: 4px;
       margin-top: 24px; /* 为标题留出空间 */
       padding-top: 10px;
       background-color: #0f121a; /* 比背景稍亮，形成层级 */
   }

   QGroupBox::title {
       subcontrol-origin: margin;
       subcontrol-position: top left;
       left: 10px;
       padding: 0 6px;
       color: #00f0ff; /* 赛博朋克核心色：青色 */
       font-weight: bold;
       background-color: #0b0f19; /* 遮挡住边框线 */
   }

   /* =======================================================
      LABELS (标签)
      稍微调暗，让数据更突出
      ======================================================= */
   QLabel {
       color: #718096;
       font-weight: normal;
   }

   /* =======================================================
      INPUT FIELDS (输入框 - 核心数据区)
      黑底、亮字、底部发光线条
      ======================================================= */
   QLineEdit, QSpinBox, QComboBox {
       background-color: #111625;
       border: 1px solid #2d3748;
       border-bottom: 2px solid #2d3748; /* 默认底部暗色 */
       border-radius: 2px;
       color: #00f0ff; /* 输入的数据显示为高亮青色 */
       padding: 6px;
       selection-background-color: #00f0ff;
       selection-color: #000000;
   }

   /* 鼠标悬停 */
   QLineEdit:hover, QSpinBox:hover, QComboBox:hover {
       background-color: #151a2b;
       border-color: #4a5568;
   }

   /* 获得焦点（输入时）：底部线条变亮，产生通电感 */
   QLineEdit:focus, QSpinBox:focus, QComboBox:focus {
       background-color: #151a2b;
       border: 1px solid #00f0ff; /* 全边框发光 */
       border-bottom: 2px solid #00f0ff;
       color: #ffffff; /* 输入时文字变白，更清晰 */
   }

   /* 下拉框弹出的列表样式 */
   QComboBox QAbstractItemView {
       background-color: #111625;
       color: #00f0ff;
       border: 1px solid #2d3748;
       selection-background-color: #2d3748;
       selection-color: #00f0ff;
   }

   /* =======================================================
      RADIO  CHECKBOX (单选/复选框)
      自定义绘制，去除系统默认的土气样式
      ======================================================= */
   QRadioButton, QCheckBox {
       spacing: 8px;
       color: #a0a8b7;
   }

   QRadioButton::indicator {
       width: 14px;
       height: 14px;
       border-radius: 7px;
       border: 1px solid #4a5568;
       background-color: #111625;
   }

   QRadioButton::indicator:checked {
       border: 2px solid #00f0ff;
       background-color: #00f0ff; /* 选中变实心青色 */
   }

   QCheckBox::indicator {
       width: 14px;
       height: 14px;
       border-radius: 2px;
       border: 1px solid #4a5568;
       background-color: #111625;
   }

   QCheckBox::indicator:checked {
       border: 1px solid #00f0ff;
       background-color: #00f0ff;
   }



/* =======================================================
   BUTTONS (点击Start后彻底熄灭)
   ======================================================= */

QPushButton {
    border: none;
    border-radius: 2px;
    padding: 12px;
    font-weight: bold;
    font-size: 14px;
    letter-spacing: 1px;
}

/* --- START 按钮 (正常状态：亮绿) --- */
QPushButton#btnStartSend {
    background-color: rgba(0, 230, 118, 0.1);
    border: 1px solid #00e676;
    color: #00e676;
}

QPushButton#btnStartSend:hover {
    background-color: rgba(0, 230, 118, 0.2);
    border: 1px solid #ffffff;
    color: #ffffff;
}

/* --- [修改点] START 按钮 (禁用状态：完全熄灭) --- */
/* 这里直接使用了和 STOP 按钮禁用态一样的颜色，彻底变灰 */
QPushButton#btnStartSend:disabled {
    background-color: transparent; /* 透明背景 */
    border: 1px solid #2d3748;     /* 暗灰边框 */
    color: #4a5568;                /* 暗灰文字 */
}


/* --- STOP 按钮 (正常状态：警报红) --- */
QPushButton#btnStopSend {
    background-color: rgba(255, 23, 68, 0.1);
    border: 1px solid #ff1744;
    color: #ff1744;
}

QPushButton#btnStopSend:hover {
    background-color: #ff1744;
    color: #ffffff;
}

/* --- STOP 按钮 (禁用状态：隐形/暗淡) --- */
QPushButton#btnStopSend:disabled {
    background-color: transparent;
    border: 1px solid #2d3748;
    color: #4a5568;
}
   </string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_Main">
    <property name="spacing">
     <number>20</number>
    </property>
    <property name="leftMargin">
     <number>20</number>
    </property>
    <property name="topMargin">
     <number>20</number>
    </property>
    <property name="rightMargin">
     <number>20</number>
    </property>
    <property name="bottomMargin">
     <number>20</number>
    </property>
    <item>
     <widget class="QGroupBox" name="grpSender">
      <property name="minimumSize">
       <size>
        <width>420</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>420</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Sender Mode</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_Left">
       <property name="spacing">
        <number>15</number>
       </property>
       <item>
        <widget class="QLabel" name="lblTxInt">
         <property name="text">
          <string>Interface Selection:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboInterfaceTx">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>35</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="grpAddr">
         <property name="title">
          <string>Address &amp; Protocol</string>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <property name="verticalSpacing">
           <number>12</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="l1">
            <property name="text">
             <string>Src MAC:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="editSrcMac">
            <property name="text">
             <string>C8:A3:62:21:99:14</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="l2">
            <property name="text">
             <string>Dst MAC:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="editDstMac">
            <property name="text">
             <string>50:D2:F5:BE:20:3A</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="l3">
            <property name="text">
             <string>Src IP:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="editSrcIp">
            <property name="text">
             <string>192.168.1.203</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="l4">
            <property name="text">
             <string>Dst IP:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="editDstIp">
            <property name="text">
             <string>192.168.1.1</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="grpParam">
         <property name="title">
          <string>Parameters</string>
         </property>
         <layout class="QVBoxLayout" name="vlParam">
          <property name="spacing">
           <number>10</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="hlProto">
            <item>
             <widget class="QRadioButton" name="rbIcmp">
              <property name="text">
               <string>ICMP</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="rbUdp">
              <property name="text">
               <string>UDP</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="rbTcp">
              <property name="text">
               <string>TCP</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="rbDns">
              <property name="text">
               <string>DNS</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QFrame" name="containerTcpFlags">
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Plain</enum>
            </property>
            <layout class="QHBoxLayout" name="hlTcpFlags">
             <property name="spacing">
              <number>15</number>
             </property>
             <property name="topMargin">
              <number>5</number>
             </property>
             <property name="bottomMargin">
              <number>5</number>
             </property>
             <item>
              <widget class="QLabel" name="lblFlags">
               <property name="styleSheet">
                <string>color: #E0E0E0; font-weight: bold;</string>
               </property>
               <property name="text">
                <string>Flags:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="chkSyn">
               <property name="cursor">
                <cursorShape>PointingHandCursor</cursorShape>
               </property>
               <property name="text">
                <string>SYN</string>
               </property>
               <property name="checked">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="chkAck">
               <property name="cursor">
                <cursorShape>PointingHandCursor</cursorShape>
               </property>
               <property name="text">
                <string>ACK</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="chkPsh">
               <property name="cursor">
                <cursorShape>PointingHandCursor</cursorShape>
               </property>
               <property name="text">
                <string>PSH</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="chkFin">
               <property name="cursor">
                <cursorShape>PointingHandCursor</cursorShape>
               </property>
               <property name="text">
                <string>FIN</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="chkRst">
               <property name="cursor">
                <cursorShape>PointingHandCursor</cursorShape>
               </property>
               <property name="text">
                <string>RST</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="spacerFlags">
               <property name="orientation">
                <enum>Qt::Orientation::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="containerDns">
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Plain</enum>
            </property>
            <layout class="QHBoxLayout" name="hlDns">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>5</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>5</number>
             </property>
             <item>
              <widget class="QLabel" name="lblDomain">
               <property name="styleSheet">
                <string>color: #E0E0E0; font-weight: bold;</string>
               </property>
               <property name="text">
                <string>Domain:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="editDomain">
               <property name="styleSheet">
                <string>background-color: #2D2D30; color: #F0F0F0; border: 1px solid #3E3E42; padding: 4px;</string>
               </property>
               <property name="text">
                <string>www.google.com</string>
               </property>
               <property name="placeholderText">
                <string>e.g. example.com</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="grpPayload">
            <property name="title">
             <string>Payload Options</string>
            </property>
            <layout class="QVBoxLayout" name="vlPayload">
             <property name="spacing">
              <number>8</number>
             </property>
             <item>
              <layout class="QHBoxLayout" name="hlPayMode">
               <item>
                <widget class="QRadioButton" name="rbPayRandom">
                 <property name="text">
                  <string>Random</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="rbPayFixed">
                 <property name="text">
                  <string>Fixed Byte</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="rbPayCustom">
                 <property name="text">
                  <string>Custom String</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="hlPayDetail">
               <item>
                <widget class="QLabel" name="lblLen">
                 <property name="text">
                  <string>Len:</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="spinPktLen">
                 <property name="minimum">
                  <number>0</number>
                 </property>
                 <property name="maximum">
                  <number>65500</number>
                 </property>
                 <property name="value">
                  <number>64</number>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="lblFixVal">
                 <property name="text">
                  <string>Val(Hex):</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="spinFixVal">
                 <property name="displayIntegerBase">
                  <number>16</number>
                 </property>
                 <property name="prefix">
                  <string>0x</string>
                 </property>
                 <property name="maximum">
                  <number>255</number>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="editCustomData">
                 <property name="placeholderText">
                  <string>Enter text payload...</string>
                 </property>
                 <property name="visible">
                  <bool>false</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <layout class="QFormLayout" name="formLayout_Param">
            <property name="labelAlignment">
             <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="lblIntVal">
              <property name="text">
               <string>Interval (µs):</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="spinInterval">
              <property name="maximum">
               <number>10000000</number>
              </property>
              <property name="singleStep">
               <number>1000</number>
              </property>
              <property name="value">
               <number>10000</number>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="lblSPort">
              <property name="text">
               <string>Src Port:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="spinSrcPort">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>65535</number>
              </property>
              <property name="value">
               <number>10086</number>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="lblDPort">
              <property name="text">
               <string>Dst Port:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="spinDstPort">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>65535</number>
              </property>
              <property name="value">
               <number>10086</number>
              </property>
             </widget>
            </item>
            </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="hlBtnTx">
         <item>
          <widget class="QPushButton" name="btnStartSend">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>50</height>
            </size>
           </property>
           <property name="text">
            <string>START SENDING</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btnStopSend">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>50</height>
            </size>
           </property>
           <property name="text">
            <string>STOP</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="spacerLeft">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
